### DISCUSSION TITLE
Subscriptions as Purchasables?

### DISCUSSION THREAD

**Discussion by nateiler** [craftcms/commerce-stripe#163] [Ideas]
I'm sure you've given it a lot of thought which is why a subscription is processed outside the typical cart flow, but can you outline some of the reasons why?  I'm keep wishing a subscription could be added to the cart and handled similar to the checkout process.  

Obviously not expecting this as a feature, but also don't want to waste a lot of time if there are obstacles that I'm not thinking of.

**Comment by andris-sevcenko** [comment]
It would be the Plan that is a Purchasable, but yeah, it's on the list of things to do. It'll be there before we release Commerce 2 GA. I'll leave this one open so that I have the satisfaction of closing it when it's time.

**Comment by nateiler** [comment]
Awesome.  Good to know.

Yes, plans.  We're on the same page 👍

**Comment by bryantAXS** [comment]
@nateiler fwiw, it's pretty easy to do this manually with a module/plugin. You can hook onto the EVENT_AFTER_COMPLETE_ORDER and then use the subscription service to manually add a user to the subscription. We ended up creating some "subscription products" that had a custom field indicating the handle of the subscription it should be linked too.

**Comment by nateiler** [comment]
@bryantAXS Awesome.  I was thinking along the same lines.  I didn't dive into what it would take to make the actual subscription purchasable, but for now I'll save some time and wait for first party development.

**Comment by bymayo** [comment]
We've also got this problem at the moment. Where the client wants users to be able to subscribe to multiple subscriptions, see a total and also add a discount code. So a standard cart would work better than individual checkout pages.

**Comment by AugustMiller** [comment]
☝️ Watch out! Discounts may not work as expected, since subscriptions are beholden to pre-configured plans in Stripe…
